Automattic/wp-calypso
# Add cards for authorizing and connecting to GitHub

**Update**: the changes included in this PR now go beyond the scope of the original, so I have made it a draft. I will likely abandon it in favour of some smaller PRs later. 

---

Closes https://github.com/Automattic/wp-calypso/issues/70376

See also D98139-code

#### Proposed Changes

* Add a new card to the hosting page that can be used to start the GitHub auth flow. 

![image](https://user-images.githubusercontent.com/6851384/212680879-82d22796-5855-4a28-99da-101869dc5e61.png)

 Notes:
-  this card is always shown for now, but it should be shown conditionally according to if we are already connected or not
- The copy isn't final. Proposals welcome
- It implements the mockup UI, but we will want to link to docs and stuff in the final version (docs update is covered by another task)
- It uses the WordPress app, which is owned by my account currently but will be transferred to Automattic later https://github.com/apps/wordpress-dot-com



#### Testing Instructions



* Apply patch D98139-code
* Make sure the API is sandboxed
* Checkout this branch local
* Go to "Hosting configuration" for an Atomic site
* Click authorise and you should be redirected to somewhere like https://public-api.wordpress.com/rest/v1.2/github-authorize-callback?siteId=210341954&code=0922b1371c663ecc5092, which is expected until more endpoints are added. 
* Verify the `siteId` and `code` are present in the URL

#### Pre-merge Checklist





Related to #
